### PR TITLE
Minor CI fix

### LINKED ISSUE #2584: Change the default keys for jsoc search results to be T_REC rather than DATE and T_OBS
This is as simple as changing [this](https://github.com/sunpy/sunpy/blob/master/sunpy/net/jsoc/jsoc.py#L843) line to read:
```python
['T_REC', 'TELESCOP', 'INSTRUME', 'WAVELNTH']
```
Unless @mbobra has a better idea.

### PR DESCRIPTION
**Pull request by nabobalis**
DO NOT MERGE THIS (YET)

Closes #2584

### CODE REVIEW

**Comment by nabobalis** [comment]
Interested to see how the Docs look without that before the theme updates.

**Comment by Cadair** [comment]
the sidebar broke.

**Comment by nabobalis** [comment]
I saw. I need to update the theme and release before we merge this.

**Comment by Cadair** [comment]
oh you broke it! ^^

File: requirements/rtd.txt [hunk 1]
```diff
@@ -1,4 +1,10 @@
 
+# Special version for RTD.
+
 -r docs.txt
```
Comment (Cadair) [inline requirements/rtd.txt]: this dosen't work, we should pull it and replace it with the base and dev

Comment (nabobalis) [reply requirements/rtd.txt]: Sure, have to merge to find out!

File: requirements/rtd.txt [hunk 2]
```diff
@@ -1,4 +1,8 @@
 
--r docs.txt
```
Comment (Cadair) [inline requirements/rtd.txt]: we still need to include `-r base.txt` and `-r dev.txt` as well.

Comment (nabobalis) [reply requirements/rtd.txt]: GRUMBLE GRUMBLE

File: setup.py [hunk 3]
```diff
@@ -121,7 +121,8 @@
                         'astropy>=2.0.3',
                         'scipy',
                         'pandas>=0.12.0',
-                        'matplotlib>=1.3'],
+                        'matplotlib>=1.3',
+                        'pytest'],  # Upstream bug fix
```
Comment (dstansby) [inline setup.py]: I don't think forcing users to install pytest is a great idea, maybe it could be added to `extras_require`?

Comment (Cadair) [reply setup.py]: unfortunately as it's currently a hard dependency for the functioning of `coordinates` I am not really happy with people potentially not having it.

Comment (dstansby) [reply setup.py]: All the other stuff in `extras_require` are hard dependencies for other modules though. (the background here is I want to use some of the coordinate stuff in HelioPy, but don't want to depend on SunPy if it means lots of other dependencies, e.g. AstroPy only requires numpy to install)

**Comment by nabobalis** [comment]
Why is it?

**Comment by Cadair** [comment]
Because that re-arrangement of jsoc keys is going to change every single output of search results from the jsoc client.

**Comment by nabobalis** [comment]
IT READY TO GO

**Pull request merged**